lib/system.nim

```diff
@@ -2651,19 +2651,40 @@ proc compiles*(x: untyped): bool {.magic: "Compiles", noSideEffect, compileTime.
   ##     echo "'+' for integers is available"
   discard
 
+proc isNamedTuple*(T: type): bool =
+  ## return true for named tuples, false for any other type.
+  when T isnot tuple: result = false
+  else:
+    var t: T
+    for name, _ in t.fieldPairs:
+      when name == "Field0":
+        return compiles(t.Field0)
```

Comment:
Huh? When would that not ``compiles``?

Reply:
that's the hack I'm using to detect named vs un-named tuple reliably:
```nim
  let a1 = (Field0:"foo",)
  echo a1.Field0 # works
  let a2 = ("foo",)
  # echo a2.Field0 # fails
```

another option would be to add a magic.

Reply:
I don't like when anything is based on the `compiles` attribute. For my personal preference you should add a magic, but I think @Araq has a different opinion about it.